src/cargo/core/workspace.rs

```diff
@@ -1883,16 +1892,19 @@ impl WorkspaceRootConfig {
             // If glob does not find any valid paths, then put the original
             // path in the expanded list to maintain backwards compatibility.
             if expanded_paths.is_empty() {
-                expanded_list.push(pathbuf);
+                expanded_list.push((pathbuf, None));
             } else {
+                let used_glob_pattern = expanded_paths.len() > 1 || expanded_paths[0] != pathbuf;
```

Comment:
Why the `>1` check?

Reply:
When one path expands to multiple paths, that's certain it's due to a glob.